scripts/bash_lib.sh

```diff
@@ -528,4 +537,133 @@ check_upgraded_versions() {
     bb_log_err "or if it is a development tree that is based on an old version"
     exit 1
   fi
+
+  res=0
+  engines_disappeared_or_changed=$(comm -23 ./engines.old ./engines.new | wc -l)
+  if ((engines_disappeared_or_changed != 0)); then
+    bb_log_err "the lists of engines in the old and new installations differ"
+    diff -u ./engines.old ./engines.new
+    res=1
+  fi
+  if [[ $test_type == "minor" ]]; then
+
+    # We are using temporary files for further comparison, because we will need to make
+    # some adjustments to them to avoid false positives, but we want original files
+    # to be stored by buildbot
+    for f in ldd-main ldd-columnstore reqs-main reqs-columnstore plugins capabilities; do
+      if [ -e "./${f}.old" ] ; then
+        cp "./${f}.old" "./${f}.old.cmp"
+        cp "./${f}.new" "./${f}.new.cmp"
+      fi
+    done
+
+    # Permanent adjustments
+    sed -i '/libstdc++.so.6(GLIBCXX/d;/libstdc++.so.6(CXXABI/d' ./reqs-*.cmp
+    # Don't compare subversions of libsystemd.so (e.g. libsystemd.so.0(LIBSYSTEMD_227)(64bit))
+    sed -i '/libsystemd.so.[0-9]*(LIBSYSTEMD_/d' ./reqs-*.cmp
+    # Ignore shells, the number of which always changes
+    sed -i '/^\/bin\/sh$/d' ./reqs-*.cmp
+
+    # Here is the place for temporary adjustments,
+    # when legitimate changes in dependencies happen between minor versions.
+    # The adjustments should be done to .cmp files, and removed after the release
+    #
+    # End of temporary adjustments
+
+    bb_log_info "Comparing old and new ldd output for installed binaries"
+    # We are not currently comparing it for Columnstore, too much trouble for nothing
+    if ! diff -U1000 ./ldd-main.old.cmp ./ldd-main.new.cmp | (grep -E '^[-+]|^ =' || true) ; then
+      bb_log_error "Found changes in ldd output on installed binaries"
```

Comment:
`sed 's/bb_log_error/bb_log_err/g'`

Reply:
fixed.

> if you want to test those scripts manually, you can call the script on a clean VM 

That's a great capability, helps a lot, thanks.
Checked:
- minor and major upgrade on alma8-amd64 10.5 (minor upgrade fails, but on the "right" reasons, due to differences in dependencies and requirements)
- minor and major upgrade on ubuntu2004-amd64 10.5 (it passes as far as scripts are concerned, but I noticed that the check for mysql_upgrade is fragile, on my very slow VM the test failed several times as it just didn't wait long enough. We'll need to see if it happens in the buildbot, but since it's an old check, it is out of the scope of this PR)
- minor upgrade on fedora39-amd64 10.6 (failed due to a requirement difference)